File: Java/src/StudentsDatabaseApp/Student.java
package StudentsDatabaseApp;

import java.util.ArrayList;
import java.util.Scanner;

public class Student
{
    private String firstName;
    private String lastName;
    private int eduYear;
    private String studentID;
    private ArrayList<String> courses = new ArrayList<String>();
    private int balance;
    private static int cost = 600;
    private static int id = 1000;

    public Student()
    {
        Scanner in = new Scanner(System.in);
        System.out.print("\nPlease enter student's first name: ");
        this.firstName = in.nextLine();

        System.out.print("Please enter student's last name: ");
        this.lastName = in.nextLine();

        System.out.print("\nPlease enter student's year of education:\n1 - Freshman;\n2 - Sophmore;\n3 - Junior;\n4 - Senior;\n");
            System.out.print("Student's year is: ");
            int temp = 0;
            while (temp == 0)
            {
                try
                {
                    eduYear = Integer.parseInt(in.nextLine());
                    if (eduYear < 1 || eduYear > 4)
                    {
                        while (eduYear < 1 || eduYear > 4)
                        {
                            System.out.print("Error! Please enter number from 1 to 4: ");
                            this.eduYear = in.nextInt();
                        }
                    }
                    temp = 1;
                }
                catch (Exception e)
                {
                    System.out.print("Please enter a number: ");
                }
            }    
        System.out.println("");
        setStudentID();
        in.close();
    }

    private void setStudentID()
    {
        id++;
        this.studentID = eduYear + "" + id;
    }

    public void enroll()
    {
        Scanner in = new Scanner(System.in);
        do
        {
            System.out.print("Please enter course to enroll or Q to quit: ");
            String course = in.nextLine();
            if (!course.equals("Q") && !course.equals("q"))
            {
                courses.add(course);
                balance += cost;
            }
            else 
            {
                System.out.println("Selection completed.\n");
                break;
            }
        } while (1 != 0);
        in.close();
    }

    public void viewBalance()
    {
        System.out.println("Student's debt is: " + balance + "$");
    }

    public void coursesPayment()
    {
        int temp =0;
        viewBalance();
        if (balance < 600)
        {
            System.out.println("Student has no debts.");
        }
        else
        {
            System.out.print("Please enter sum student want to pay: ");
            while (temp == 0)
            {
                try
                {
                    Scanner in = new Scanner(System.in);
                    int payment = Integer.parseInt(in.nextLine());
                    if (payment > balance)
                    {
                        while (payment > balance)
                        {
                            System.out.print("Please enter number less than your debt: ");
                            payment = Integer.parseInt(in.nextLine());
                        }
                    }
                    else if (payment <= -1)
                    {
                        while (payment < 0)
                        {
                            System.out.print("Please enter positive number: ");
                            payment = Integer.parseInt(in.nextLine());
                        }
                    }
                    balance -= payment;
                    System.out.print("Thank you for your payment!\nYour new balance is: ");
                    viewBalance();
                    temp = 1;
                    in.close();
                }
                catch (Exception e)
                {
                    System.out.print("Please enter sum you want to pay: ");
                }
            }  
        }
    }

    public String showInfo()
    {
        return "\n~~~Student Information~~~" + "\nName: " + firstName + " " + lastName + ";" + "\nStudent ID: " + studentID + ";" + "\nCourses enrolled: " + courses + "\nYour debt: " + balance + "$";
    }
}

File: C#/Student/Student.cs
using System.Collections;

class Student
{
    private string firstName;
    private string lastName;
    private int eduYear;
    private string studentId;
    private ArrayList courses = new ArrayList();
    private int balance;
    private int cost = 600;
    private int id = 1000;
    bool temp = true;

    public Student()
    {
        Console.WriteLine("Please enter student's first name: ");
        this.firstName = Console.ReadLine();
        Console.WriteLine("Please enter student's last name: ");
        this.lastName = Console.ReadLine();
        Console.WriteLine("\nPlease enter student's year of education:\n1 - Freshman;\n2 - Sophmore;\n3 - Junior;\n4 - Senior;\n");
        Console.WriteLine("Student's year is: ");
        while (temp == true)
        {
            try
            {   
                eduYear = Int32.Parse(Console.ReadLine());
                switch (eduYear, eduYear, eduYear, eduYear)
                {
                    case (1, 2, 3, 4):
                        temp = false;
                        break;
                    default:
                        Console.WriteLine("Please enter number from 1 to 4: ");
                        break;
                }
            }
            catch (Exception e)
            {
                Console.WriteLine("Please enter a number: ");
            }
        }
        setStudentId();
    }

    private void setStudentId()
    {
        id++; 
        this.studentId = eduYear + "" + id;
    } 
    
    public void enroll()
    {
        do 
        {
            Console.WriteLine("Please enter course to enroll or Q to quit: ");
            string course = Console.ReadLine();
            if (course != "q" || course != "Q")
            {
                courses.Add(course);
                balance += cost;
            }
            else
            {
                Console.WriteLine("Selection completed.\n");
                temp = false;
            }
        }
        while(temp == true);
    }

    public void viewBalance()
    {
        Console.WriteLine("Student's debt is: " + balance + "$");
    }

    public void coursesPayment()
    {
        viewBalance();
        if (balance < 600)
        {
            Console.WriteLine("Student has no debts.");
        }
        else
        {
            Console.WriteLine("Please enter sum student want to pay: ");
            while (temp == true)
            {
                try
                {
                    int payment = Int32.Parse(Console.ReadLine());
                    if (payment > balance)
                    {
                        while (payment > balance)
                        {
                            Console.WriteLine("Please enter number less than your debt: ");
                            payment = Int32.Parse(Console.ReadLine());
                        }
                    }
                    else if (payment <= -1)
                    {
                        while (payment < 0)
                        {
                            Console.WriteLine("Please enter positive number: ");
                            payment = Int32.Parse(Console.ReadLine());
                        }
                    }
                    balance -= payment;
                    Console.WriteLine("Thank you for your payment!\nYour new balance is: ");
                    viewBalance();
                    temp = false;
                }
                catch (Exception e)
                {
                    Console.WriteLine("Please enter sum you want to pay: ");
                }
            }  
        }
    }

    public String showInfo()
    {
        return "\n~~~Student Information~~~" + "\nName: " + firstName + " " + lastName + ";" + "\nStudent ID: " + studentId + ";" + "\nCourses enrolled: " + courses + "\nYour debt: " + balance + "$";
    }
}